src/python/WMCore/BossAir/Plugins/PyCondorPlugin.py

```diff
@@ -1001,15 +1008,21 @@ def customizePerJob(self, job):
         # Prefer slots that are closest to our MaxCores without going over.
         # If the slot size is _greater_ than our MaxCores, we prefer not to
         # use it - we might unnecessarily fragment the slot.
-        jdl.append('rank = ifThenElse(Cpus > MaxCores, -Cpus, Cpus)\n')
+        jdl.append('rank = isUndefined(Cpus) ? 0 : ifThenElse(Cpus > MaxCores, -Cpus, Cpus)\n')
         # Record the number of CPUs utilized at match time.  We'll use this later
         # for monitoring and accounting.  Defaults to 0; once matched, it'll
         # put an attribute in the job  MATCH_EXP_JOB_GLIDEIN_Cpus = 4
         jdl.append('+JOB_GLIDEIN_Cpus = "$$(Cpus:0)"\n')
         # Make sure the resize request stays within MinCores and MaxCores.
         jdl.append('+RequestResizedCpus = (Cpus>MaxCores) ? MaxCores : ((Cpus < MinCores) ? MinCores : Cpus)\n')
+        # Advertise the original CPU setting, in case someone needs this for monitoring
+        jdl.append('+OriginalCpus = %d\n' % origCores)
+        # If the job is running, then we should report the matched CPUs in RequestCpus - but only if there are sane
+        # values.  Otherwise, we just report the original CPU request
+        jdl.append('+JobCpus = ((JobStatus =!= 1) && (JobStatus =!= 5) && !isUndefined(MATCH_EXP_JOB_GLIDEIN_Cpus) && (int(MATCH_EXP_JOB_GLIDEIN_Cpus) isnt error)) ? int(MATCH_EXP_JOB_GLIDEIN_Cpus) : OriginalCpus\n')
```

Comment:
Yeah - I don't want things to go haywire because some future person sets the value to "foo" instead of "1".  If it's not parseable, we don't want to use its value.

You can now see why I broke up the logic for `RequestCpus` into several expressions :)